### PR TITLE
python3Packages.cantools: init at 38.0.2

### PR DESCRIPTION
**Pull request by gray-heron**
###### Description of changes

 - Added python-based utilities for the CAN bus: https://github.com/cantools/cantools
 - Added textparser python packge (https://github.com/eerimoq/textparser), as `cantools`  depends on it.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by simoneruffini** [comment]
Do we need to run all the tests of cantools and textparser when building the packages?
Moreover works for me.

**Comment by simoneruffini** [comment]
I wasn't able to plot data with matplotlib, does it work on your end?

**Comment by gray-heron** [comment]
> Do we need to run all the tests of cantools and textparser when building the packages?

I'm not sure what is the common practice, but I see tests are being run on some other Python packages. 


> I wasn't able to plot data with matplotlib, does it work on your end?

Good catch. Fixed.

**Comment by simoneruffini** [comment]
Works for me, rebase the commits into one so we can proceed

**Comment by gray-heron** [comment]
Done.

**Review by simoneruffini — APPROVED** [review]
wfm

File: maintainers/maintainer-list.nix [hunk 1]
```diff
@@ -6333,6 +6333,12 @@
     githubId = 4656860;
     name = "Gaute Ravndal";
   };
+  gray-heron = {
+    email = "ave+nix@cezar.info";
+    github = "gray-heron";
+    githubId = 7032646;
+    name = "Cezary Siwek";
+  };
```
Comment (pbsds) [inline maintainers/maintainer-list.nix:6341]: Please split this change into a commit titled `maintainers: add gray-heron`

File: pkgs/development/python-modules/cantools/default.nix [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+    pytestCheckHook
+  ];
+
+  pythonImportsCheck = [
+    "cantools"
+  ];
+
+  meta = with lib; {
+    homepage = "https://github.com/cantools/cantools";
+    description = "CAN bus tools.";
```
Comment (pbsds) [inline pkgs/development/python-modules/cantools/default.nix:54]:
```suggestion
    description = "CAN bus tools";
```

File: pkgs/development/python-modules/textparser/default.nix [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+    inherit pname version;
+    hash = "sha256-VvcI51qp0AKtt22CO6bvFm1+zsHj5MpMHKED+BdWgzU=";
+  };
+
+  nativeBuildInputs = [
+    setuptools-scm
+  ];
+
+  propagatedBuildInputs = [
+  ];
```
Comment (pbsds) [inline pkgs/development/python-modules/textparser/default.nix:26]:
```suggestion
```

File: pkgs/development/python-modules/cantools/default.nix [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+  ];
+
+  propagatedBuildInputs = [
+    argparse-addons
+    bitstruct
+    can
+    crccheck
+    diskcache
+    matplotlib
+    parameterized
```
Comment (pbsds) [inline pkgs/development/python-modules/cantools/default.nix:40]: Please move to native check inputs

```suggestion
```

**Review by pbsds — CHANGES_REQUESTED** [review]
Great work, welcome to nixpkgs!

Please split `cantools` and `textparser` into two separate commits, titled according to [convention](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#commit-conventions):

* `python3Packages.textparser: init at 0.24.0`
* `python3Packages.cantools: init at 38.0.2`

---

- [x] package path fits guidelines
- [x] package name fits guidelines
- [x] package version fits guidelines
- [x] package build on x86_64-linux
- [x] executables tested on x86_64-linux
- [ ] `meta.description` is set and fits guidelines
- [x] `meta.license` fits upstream license
- [x] `meta.platforms` is set
- [x] `meta.maintainers` is set
- [ ] build time only dependencies are declared in `nativeBuildInputs`
- [x] source is fetched using the appropriate function
- [x] the list of `phases` is not overridden
- [x] when a phase (like `installPhase`) is overridden it starts with `runHook preInstall` and ends with `runHook postInstall`.
- [x] patches have a comment describing either the upstream URL or a reason why the patch wasn't upstreamed
- [x] patches that are remotely available are fetched rather than vendored


---
Result of `nixpkgs-review pr 232392` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>8 packages built:</summary>
  <ul>
    <li>python310Packages.cantools</li>
    <li>python310Packages.cantools.dist</li>
    <li>python310Packages.textparser</li>
    <li>python310Packages.textparser.dist</li>
    <li>python311Packages.cantools</li>
    <li>python311Packages.cantools.dist</li>
    <li>python311Packages.textparser</li>
    <li>python311Packages.textparser.dist</li>
  </ul>
</details>

**Comment by gray-heron** [comment]
Thanks, done.

File: pkgs/development/python-modules/textparser/default.nix [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+    pytestCheckHook
+  ];
+
+  pythonImportsCheck = [
+    "textparser"
+  ];
+
+  meta = with lib; {
+    homepage = "https://github.com/eerimoq/textparser";
+    description = "A text parser.";
```
Comment (pbsds) [inline pkgs/development/python-modules/textparser/default.nix:35]:
```suggestion
    description = "A text parser";
```

**Review by pbsds — APPROVED** [review]
Just a minor nit left.
BTW, if you change the pull title to `python3Packages.cantools: init at xyz` then a nixpkgs comitter is more likely to stroll by ;)

---



- [x] package path fits guidelines
- [x] package name fits guidelines
- [x] package version fits guidelines
- [x] package build on x86_64-linux
- [x] executables tested on x86_64-linux
- [x] `meta.description` is set and fits guidelines
- [x] `meta.license` fits upstream license
- [x] `meta.platforms` is set
- [x] `meta.maintainers` is set
- [x] build time only dependencies are declared in `nativeBuildInputs`
- [x] source is fetched using the appropriate function
- [x] the list of `phases` is not overridden
- [x] when a phase (like `installPhase`) is overridden it starts with `runHook preInstall` and ends with `runHook postInstall`.
- [x] patches have a comment describing either the upstream URL or a reason why the patch wasn't upstreamed
- [x] patches that are remotely available are fetched rather than vendored

---

Result of `nixpkgs-review pr 232392` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>8 packages built:</summary>
  <ul>
    <li>python310Packages.cantools</li>
    <li>python310Packages.cantools.dist</li>
    <li>python310Packages.textparser</li>
    <li>python310Packages.textparser.dist</li>
    <li>python311Packages.cantools</li>
    <li>python311Packages.cantools.dist</li>
    <li>python311Packages.textparser</li>
    <li>python311Packages.textparser.dist</li>
  </ul>
</details>

**Comment by gray-heron** [comment]
Done & done, thanks for the tip!

**Review by pbsds — APPROVED** [review]

**Pull request merged**